### PR TITLE
Remove coreir.BitVector and use the BitVector class from the bit_vector package

### PR DESCRIPTION
**Pull request by shacklettbp**
Allows for a better API for constructing BitVectors.

Requires changes to any package using coreir.BitVector since the width and init arguments are flipped in the bit_vector package.

### CODE REVIEW

File: coreir/type.py [hunk 1]
```diff
@@ -60,11 +56,11 @@ def value(self):
                 prefix, value = value_str.value.split("'h")
                 value = int(value, 16)
 
-                return BitVector(width.value, value)
+                return BitVector(value, num_bits=width.value)
             else:
                 width = ct.c_int()
                 libcoreir_c.COREValueBitVectorGetWidth(self.ptr, ct.byref(width))
-                return BitVector(width.value)
+                return BitVector(0, num_bits=width.value)
```
Comment (leonardt) [inline coreir/type.py]: In this case, it should actually return a BitVector with an `x` (unknown value). Not sure about the best way to handle this, one option would be to extend the `bit_vector.BitVector` class to support initialization with the value `None` which is a BitVector object with an uninitialized value. One interesting behavior would be to support `and` and `or` with this object (for example `x & 0 = 0` and `x | 1 = 1`).

**Comment by leonardt** [comment]
@cdonovick (SMT-PNR) and @cristian-mattarei @makaimann, this change will likely break your usage of `pycoreir` (this pull request is going into `dev`, so you'll have time to catch up to it). Are there any issues you'd like to raise before this gets pulled in?

Basically, instead of having a one-off `BitVector` class in `pycoreir` (which is a really basic class with a `value` and `width` field), it updates it to use the BitVector defined in https://github.com/leonardt/bit_vector which has support for various operations (it was actually designed to match coreir and therefor smt primitives).

**Comment by makaimann** [comment]
@leonardt -- Nope, that all sounds good. Thanks for the heads up! We'll sync up on dev.

File: coreir/type.py (hunk 1 above)
Comment (leonardt) [reply coreir/type.py]: That seems reasonable. One thing we could consider would be defining a sub class or superclass of BitVector (maybe UndefinedBitVector?) that implements the various methods to just throw an exception. Or we can modify the BitVector __init__ to take in a value of `None` and either redefine the methods to throw an exception, or wrap the methods in a decorator which checks for a None value and throws and exception.

Comment (leonardt) [reply coreir/type.py]: @shacklettbp by the way, I think for now it's sufficient to just add support for a non binary bitvector (X) when `None` is passed in as the value, we can add support for the operators later, but if we can have a basic non binary bitvector we can get this change merged in.

**Comment by leonardt** [comment]
@cdonovick approved this change in a conversation with me today, so once we have support for `X` we can merge this in

**Pull request merged**